front/src/applications/editor/tools/rangeEdition/tool-factory.tsx

```diff
@@ -101,6 +101,16 @@ function getRangeEditionTool<T extends EditorRange>({
           async onClick({ state, setState, dispatch, infraID }) {
             const { initialEntity, entity } = state;
             if (!isEqual(entity, initialEntity)) {
+              // For each track in the speed section, check if the user dragged an extremity beyond the other one
+              // if he did, switch the values of each extremities
+              const trackRanges = entity.properties.track_ranges;
+              trackRanges?.forEach((track) => {
```

Comment:
Can you use a map ? Here you are mutating a object in an array, I'm not a big fan of mutability like this

Reply:
Here you are mutating directly a some state, you need to use setState

Reply:
Ok i see what you wanted to do.

But I think you just need to use what’s already on line 120,

entity.properties is already modified

```
newTrackRange = track_ranges.map(_=> …)

              const savedEntity =
                entityId && entityId !== entity.properties.id
                  ? {
                      ...entity,
                      properties: { ...entity.properties, id: `${entityId}`, track_ranges: newTrackRange },
                    }
                  : entity;
```

Reply:
Actually I need to do that before the api call otherwise the infra breaks if the trackranges are bad (that's the point of this issue) but I'll figure something out

Reply:
It's good for me, @anisometropie you can resolve it's fine for you too :)